Doc/reference/toplevel_components.rst

```diff
@@ -69,7 +69,7 @@ All input read from non-interactive files has the same form:
 .. grammar-snippet::
    :group: python-grammar
 
-   file_input: (NEWLINE | `statement`)*
+   file_input: (NEWLINE | `statement`)* ENDMARKER
```

Comment:
I'm not sure how these changes are related to this PR?

Reply:
The ENDMARKER docs now refer here, and it would be curious if the marker is missing.

The [actual gramar](https://github.com/python/cpython/blob/main/Grammar/python.gram) does use ENDMARKER.